cryptoki/tests/basic.rs

```diff
@@ -349,19 +350,111 @@ fn session_find_objects() {
         Attribute::KeyType(KeyType::DES3),
     ];
 
-    let mut found_keys = session.find_objects(&key_search_template).unwrap();
+    let mut found_keys = session.find_objects(&key_search_template)?;
     assert_eq!(found_keys.len(), 11);
 
     // destroy one key
-    session.destroy_object(found_keys.pop().unwrap()).unwrap();
+    session.destroy_object(found_keys.pop().unwrap())?;
 
-    let mut found_keys = session.find_objects(&key_search_template).unwrap();
+    let mut found_keys = session.find_objects(&key_search_template)?;
     assert_eq!(found_keys.len(), 10);
 
     // destroy another key
-    session.destroy_object(found_keys.pop().unwrap()).unwrap();
-    let found_keys = session.find_objects(&key_search_template).unwrap();
+    session.destroy_object(found_keys.pop().unwrap())?;
+    let found_keys = session.find_objects(&key_search_template)?;
     assert_eq!(found_keys.len(), 9);
+    Ok(())
+}
+
+#[test]
+#[serial]
+fn session_objecthandle_iterator() -> testresult::TestResult {
+    let (pkcs11, slot) = init_pins();
+    // open a session
+    let session = pkcs11.open_rw_session(slot)?;
+
+    // log in the session
+    session.login(UserType::User, Some(&AuthPin::new(USER_PIN.into())))?;
+
+    // we generate 11 keys with the same CKA_ID
+
+    (1..=11).for_each(|i| {
+        let key_template = vec![
+            Attribute::Token(true),
+            Attribute::Encrypt(true),
+            Attribute::Label(format!("key_{}", i).as_bytes().to_vec()),
+            Attribute::Id("12345678".as_bytes().to_vec()), // reusing the same CKA_ID
+        ];
+
+        // generate a secret key
+        let _key = session.generate_key(&Mechanism::Des3KeyGen, &key_template);
```

Comment:
I noticed that you removed the `unwrap` here, is it ok?

Reply:
Yeah, let's use `?` here. The warning is suppressed because of the variable assignment but better to check IMO.

Reply:
@hug-dev good catch. I will need to refactor a bit, as the closure don't like handling `?`.